node_test/reverseproxy_test.go

```diff
@@ -0,0 +1,271 @@
+	if err != nil {
+		t.Fatalf("Error creating voucher: %v", err)
+	}
+	return v
+}
+
+func expectedPaymentErrorMessage(numPaid int) string {
+	return fmt.Sprintf("payment of 5 required, the voucher only resulted in a payment of %d", numPaid)
+}
+
+// performGetRequest performs a GET request to the given url
+// If any error occurs it will fail the test
+func performGetRequest(t *testing.T, url string) *http.Response {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET",
+		url,
+		nil)
+	if err != nil {
+		t.Fatalf("Error performing request: %v", err)
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		t.Fatalf("Error performing request: %v", err)
+	}
+	return resp
+}
+
+func checkResponse(t *testing.T, resp *http.Response, expectedBody string, expectedStatusCode int) {
+	responseBodyText, statusCode := getResponseInfo(t, resp)
+	if !strings.Contains(responseBodyText, expectedBody) {
+		t.Errorf("The body of the response %s did not contain the expected text %s ", responseBodyText, expectedBody)
+	}
+	if statusCode != expectedStatusCode {
+		t.Errorf("Expected status code %d, but got %d", expectedStatusCode, statusCode)
+	}
+}
+
+// getResponseInfo reads the response body and returns it as a string
+// If any error occurs it will fail the test
+func getResponseInfo(t *testing.T, resp *http.Response) (body string, statusCode int) {
+	bodyText, err := io.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("Error reading request data: %v", err)
+	}
+	resp.Body.Close()
+
+	return string(bodyText), resp.StatusCode
+}
+
+// setupNitroClients creates three nitro clients and connects them to each other
+func setupNitroClients(t *testing.T, logDestination *os.File) (alice, irene, bob *rpc.RpcClient, cleanup func()) {
+	chain := chainservice.NewMockChain()
+	logger := testLogger(logDestination)
+
+	aliceChainService := chainservice.NewMockChainService(chain, ta.Alice.Address())
+	bobChainService := chainservice.NewMockChainService(chain, ta.Bob.Address())
+	ireneChainService := chainservice.NewMockChainService(chain, ta.Irene.Address())
+
+	aliceClient, msgAlice, aliceCleanup := setupNitroNodeWithRPCClient(t, ta.Alice.PrivateKey, 3105, 4105, aliceChainService, logDestination, "ws")
+	ireneClient, msgIrene, ireneCleanup := setupNitroNodeWithRPCClient(t, ta.Irene.PrivateKey, 3106, 4106, ireneChainService, logDestination, "ws")
+	bobClient, msgBob, bobCleanup := setupNitroNodeWithRPCClient(t, ta.Bob.PrivateKey, 3107, 4107, bobChainService, logDestination, "ws")
+
+	logger.Info().Msg("Clients created")
+
+	waitForPeerInfoExchange(msgAlice, msgBob, msgIrene)
+	logger.Info().Msg("Peer exchange complete")
+
+	return aliceClient, ireneClient, bobClient, func() {
+		aliceCleanup()
+		ireneCleanup()
+		bobCleanup()
+		chain.Close()
+	}
+}
+
+// createChannelData creates ledgers channels and a payment channel between Alice and Bob
+func createChannelData(t *testing.T, aliceClient, ireneClient, bobClient *rpc.RpcClient) (paymentChannelId types.Destination) {
```

Comment:
Is it worth it to factor out a utility given that we already have https://github.com/statechannels/go-nitro/blob/4f0c010645695f898e410f617c7cdf4d13792be3/cmd/create-channels/main.go#L37?

Reply:
I'm going to leave this as is for now, since  `createLedgerChannel` would have to be refactored a bit to be able to serve as a utility that can be used from the `create-channels` script and in tests.  We also may want to consider other utility functions; for example we also have a [node_test helper to create ledger channels](https://github.com/statechannels/go-nitro/blob/6e6231b8a7aec8f689b80bb67cbd1b609f09f4ee/node_test/helpers_test.go#L180). I'll create an issue for that.